installutils/helminstall/helm_client.go

```diff
@@ -6,11 +6,18 @@ import (
 	"github.com/spf13/afero"
 )
 
-// a HelmClient that talks to the kube api server and creates resources
-func DefaultHelmClient() types.HelmClient {
-	return &internal.DefaultHelmClient{
-		Fs:              internal.NewFs(afero.NewOsFs()),
-		ResourceFetcher: internal.NewDefaultResourceFetcher(),
-		HelmLoaders:     internal.NewHelmFactories(),
-	}
+// HelmClient factory that accepts kubeconfig as a file.
+func DefaultHelmClientFileConfigFactory() types.HelmClientForFileConfigFactory {
+	return internal.NewHelmClientForFileConfigFactory(
+		internal.NewFs(afero.NewOsFs()),
+		internal.NewDefaultResourceFetcher(),
+		internal.NewHelmFactories())
+}
+
+// HelmClient factory that accepts kubeconfig in memory.
+func DefaultHelmClientMemoryConfigFactory() types.HelmClientForMemoryConfigFactory {
+	return internal.NewHelmClientForMemoryConfigFactory(
```

Comment:
The approach in this PR seems solid overall, my only question is whether we want to embrace the factory pattern to this extent within this go-utils package. Here we have a provider for a factory that returns a helm client. Users might be happy with just `NewHelmClientForMemoryConfig`, but it's in `internal`. My preference would be to have go-utils expose simple constructors and leave the factory type definitions and implementations to importing projects, wdyt?

Reply:
Yup, that makes sense to me! I've replaced the factory constructors here with regular HelmClient constructors that expose the relevant in-mem vs. file config params